runtime/lib/app-runtime.js

```diff
@@ -77,50 +77,80 @@ AppRuntime.prototype.init = function init () {
   if (this.inited) {
     return Promise.resolve()
   }
+  /** 1. init components. */
   this.componentsInvoke('init')
+  /** 2. init device properties, such as volume and cloud stack. */
   this.initiate()
-  /** set turen to not muted */
+  this.resetCloudStack()
+  /** 3. init services */
   this.component.turen.toggleMute(false)
   this.component.turen.toggleWakeUpEngine(true)
+  this.resetServices()
 
+  /** 4. listen on lifetime events */
   this.component.lifetime.on('stack-reset', () => {
     this.resetCloudStack()
   })
   this.component.lifetime.on('preemption', appId => {
     this.appPause(appId)
   })
-  // initializing the whole process...
-  this.resetCloudStack()
-  this.resetServices()
+
+  /** 5. determines if welcome announcements are needed */
   this.shouldWelcome = !this.isStartupFlagExists()
+  var isFirstBoot = property.get('sys.firstboot.init', 'persist') !== '1'
+  property.set('sys.firstboot.init', '1', 'persist')
+
+  var shouldBreakInit = () => {
+    if (this.__temporaryDisablingReasons.indexOf('welcoming') >= 0 &&
+      this.__temporaryDisablingReasons.length > 1) {
+      return true
+    }
+    if (this.hasBeenDisabled()) {
+      return true
+    }
+    return false
+  }
 
+  /** 6. load app manifests */
   return this.loadApps().then(() => {
     this.inited = true
     this.enableRuntimeFor('initiating')
 
+    /** 7. questioning if any interests of delegation */
     return this.component.dispatcher.delegate('runtimeDidInit')
   }).then(delegation => {
     if (delegation) {
       return
     }
     this.disableRuntimeFor('welcoming')
 
+    /** 8. announce welcoming */
     var future = Promise.resolve()
```

Comment:
Could we check the break-able property here?

Reply:
No, the value has to be re-checked on every step of welcoming.

Reply:
Ok, resolved.